src/gui/transferlistwidget.cpp

```diff
@@ -208,9 +210,14 @@ inline QModelIndex TransferListWidget::mapFromSource(const QModelIndex &index) c
     return nameFilterModel->mapFromSource(index);
 }
 
-void TransferListWidget::torrentDoubleClicked(const QModelIndex& index)
+void TransferListWidget::torrentDoubleClicked()
 {
-    BitTorrent::TorrentHandle *const torrent = listModel->torrentHandle(mapToSource(index));
+    const QModelIndexList selectedIndexes = selectionModel()->selectedRows();
+    if (selectedIndexes.size() != 1) return;
+    if (!selectedIndexes.first().isValid()) return;
+
+    const QModelIndex index = listModel->index(mapToSource(selectedIndexes.first()).row());
```

Comment:
You mean pass it directly in `listModel->torrentHandle`? It'll be quite the "inception". Should i split it in 2-3 lines? Couldn't find anything in the guidelines for this and Qt creator splits the second line 12 spaces (from the previous) and the third 4 spaces. 
Something like
```
BitTorrent::TorrentHandle *const torrent = listModel->torrentHandle(
            listModel->index(
                mapToSource(selectedIndexes.first()).row()));
```